### DISCUSSION TITLE
Cannot save PNG

### DISCUSSION THREAD

**Discussion by sueng66** [python-pillow/Pillow#6380] [General]
Why I could not save Image to file ? is there any dependency ? I am using release 9.1.1, here is my code :
`im = Image.frombytes('L',(self.iwidth,self.iheight),bytes(self.ibuff))`
`im.save('/data/'+self.fname+'.png',format='PNG')`
and got error :
`KeyError: 'PNG'`

**Comment by radarhere** [comment]
There was a similar error in #5034, where the user couldn't save a PNG, and concluded the problem was that they didn't have zlib enabled.

If you run `python3 -m PIL`, you should see a bunch of information, including whether or not ZLIB is supported by your installation.

If you would just like to save an image, and don't care about the format, you could try saving a JPEG instead.

**Reply by sueng66** [reply]
happend same thing with JPEG, even with BMP, and running `python3 -m PIL` tell 
`PNG image/png`
`Extensions: .apng, .png`
`Features: open, save, save_all`

**Comment by sueng66** [comment]
I tested directly it's ok
`>>> from PIL import Image`
`>>> im = Image.frombytes('L',(2,2),b'\x00\x00\x00\x00')`
`>>> im.save('tes.png')`
it's generate tes.png and readable by other application

**Reply by radarhere** [reply]
What you mean by 'directly'? Are you running your Python code through a different application when it doesn't work?

**Reply by sueng66** [reply]
I am running above test code inside python, while the error come from my app, I think it's related with the data, still debugging it

**Comment by sueng66** [comment]
found the problem, I called from the destructor while the resource already unavailable, the solution is delete the class and then wait for a couple seconds to make sure resource still available
